src/vs/base/browser/ui/resourceviewer/resourceViewer.ts

```diff
@@ -157,9 +158,21 @@ export class ResourceViewer {
 		else {
 			$(container)
 				.empty()
-				.span({
+				.append($('p', {
 					text: nls.localize('nativeBinaryError', "The file will not be displayed in the editor because it is either binary, very large or uses an unsupported text encoding.")
-				});
+				}))
+				.append($('a', {
+					role: 'button',
+					class: 'open-external',
+					text: nls.localize('resourceOpenExternal', "Try opening file in external program?")
+				}).on(DOM.EventType.CLICK, (e) => {
+					windowsService.openExternal(descriptor.resource.toString()).then(didOpen => {
```

Comment:
Good point. I tested this and think we should restrict this to large image files for now and leave the existing flow in place for unsupported resource types.